lib/bats-core/tracing.bash

```diff
@@ -117,15 +127,61 @@ bats_trim_filename() {
 	printf -v "$2" '%s' "${1#$BATS_CWD/}"
 }
 
+# bats_debug_trap tracks the last line of code executed within a test. This is
```

Comment:
This analysis is really helpful, maybe it should be in the wiki rather than inline?

Reply:
It'd be great if you could share your methodology too, e.g. in a Gist, so we can reproduce what you're seeing.

Reply:
> It'd be great if you could share your methodology too, e.g. in a Gist, so we can reproduce what you're seeing.

Sure! Here's the script I used to inspect Bash's behavior: https://gist.github.com/strager/dd5370f27e96c0d66deebc6ab051d421